packages/algorand-js/src/interpreter/opcode-list.ts

```diff
@@ -702,7 +701,7 @@ export class EqualTo extends Op {
           stack.push(BIGINT0);
         }
       } else {
-        if (compareArray(this.assertBytes(last), this.assertBytes(prev))) {
+        if (JSON.stringify(this.assertBytes(last)) === JSON.stringify(this.assertBytes(prev))) {
```

Comment:
I think using `JSON.stringify` has some edge cases. Can you check: https://www.30secondsofcode.org/blog/s/javascript-array-comparison

Reply:
Yes, I read it, it's not the case here as we have bytes array. Also if you see one line above we are using assert for that.

But if it is being used in stateful also then it is better to have compareArray like before.

Restoring

Reply:
Also, could you restructure the code a bit? Having a waterfall of if's is hard to read. Example of improvement:
```
if (typeof last !== typeof prev)
   throw new TealError(ERRORS.TEAL.INVALID_TYPE);    // NOTE: are you sure we should throw an exception instead pushing 0 to the stack?
if (typeof last === "bigint")
   stack.push(last === prev? BIGINT1 : BIGINT0)
else {
  ...
}

Reply:
yes

![opEqq](https://user-images.githubusercontent.com/26249903/102279507-87e89680-3f51-11eb-80ac-32dde03a934d.png)